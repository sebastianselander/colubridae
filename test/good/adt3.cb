type Foo {
    Bar,
    Baz(),
}

def fun(f: fn() -> Foo) {
    let a: Foo = f();
    printString("got thing of type foo\n");
}

def main() {
    let b = Bar;
    let f = Baz;
    fun(f)
}
