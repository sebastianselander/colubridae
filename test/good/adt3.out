got thing of type foo
